Dashboard/app/js/lib/controllers/data-controllers.js

```diff
@@ -312,11 +312,78 @@ FLOW.surveyedLocaleControl = Ember.ArrayController.create({
 FLOW.questionAnswerControl = Ember.ArrayController.create({
   content: null,
 
+  // a computed property that returns a list containing *sub lists*
+  // of responses to questions. Each sub list represents a single iteration
+  // over a set of responses to questions in a specific question group, ordered
+  // by question order. For repeat question groups two adjacent sub lists
+  // represent two iterations of responses for that group
+  contentByGroup: Ember.computed('content.isLoaded', function(key, value) {
+    var content = Ember.get(this, 'content'),
+        self = this;
+    if (content) {
+      var groupQuestions = FLOW.questionControl.get('content');
+      var groups = FLOW.questionGroupControl.get('content');
+
+      var allResponses = [];
+      var groupResponses = [];
+      var group, groupId, isRepeatable, questionsInGroup, questionGroupId, questionId, answersInGroup;
+
+      for (var i = 0; i < groups.get('length'); i++) {
+        group = groups.objectAt(i);
+        isRepeatable = group.get('repeatable');
+
+        groupId = group.get('keyId');
+        questionsInGroup = groupQuestions.filterProperty('questionGroupId',groupId);
+
+        answersInGroup = self.filter(function(answer) {
+          var questionIds = questionsInGroup.map(function(question) {
+              return question.get('keyId').toString();
+          });
+          return questionIds.indexOf(answer.get('questionID')) > -1;
+        });
+
+        groupResponses.groupName = group.get('name');
+
+        if (!isRepeatable) {
+          groupResponses.push(answersInGroup)
+          allResponses.push(groupResponses);
+          groupResponses = [];
+          continue;
+        }
+
+        // determine max of iterations for this group
+        var maxIter = 0;
+        answersInGroup.forEach(function(item){
+          if (item.get('iteration') > maxIter){
+            maxIter = item.get('iteration');
+          }
+        });
+
+        for (var iter = 0; iter <= maxIter; iter++){
+          var iterationAnswers = answersInGroup.filter(function(answer) {
+            return answer.get('iteration') === iter;
+          });
+
+          // TODO: +2 is a workaround for the fact that iterations currently start at -1
```

Comment:
Is this referring to the bug we hunted yesterday? If so, all iteration IDs are guaranteed to be 0+.